optaplanner-examples/src/main/java/org/optaplanner/examples/common/business/SolutionBusiness.java

```diff
@@ -360,6 +360,8 @@ public void doProblemChange(ProblemChange<Solution_> problemChange) {
             solver.addProblemChange(problemChange);
         } else {
             problemChange.doChange(guiScoreDirector.getWorkingSolution(), problemChangeDirector);
+            // This needs to be called before the score is calculated as ProblemChange does not trigger listeners.
+            guiScoreDirector.triggerVariableListeners();
```

Comment:
Looks like a functional regression, doesn't it? Shouldn't this be fixed in the API instead?

Reply:
I don't think this is a regression. I mean, it _is_ a regression in the examples app but there is no regression for user that are using the **public** part of the problem change API itself.

Problem changes are meant for real-time planning and they are normally supposed to be submitted through `Solver` (while it's solving) in which case variable listeners are triggered automatically.

Conversely, this piece of the examples code relies on internal APIs (like `InnerScoreDirector.calculateScore()`).

Having said that, I don't see a problem in the public part of the problem change API but while I was writing this comment, I found a way to improve the internal part: https://github.com/kiegroup/optaplanner/pull/1822